packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -314,15 +314,35 @@ export class NodeProxyTarget extends ProxyTarget<Anchor> {
 		eventName: K,
 		listener: EditableTreeEvents[K],
 	): () => void {
-		assert(eventName === "changing", 0x5b3 /* unexpected eventName */);
-		const unsubscribeFromValueChange = this.anchorNode.on("valueChanging", () => listener());
-		const unsubscribeFromChildrenChange = this.anchorNode.on("childrenChanging", () =>
-			listener(),
-		);
-		return () => {
-			unsubscribeFromValueChange();
-			unsubscribeFromChildrenChange();
-		};
+		switch (eventName) {
+			case "changing": {
+				const unsubscribeFromValueChange = this.anchorNode.on(
+					"valueChanging",
+					(anchorNode: AnchorNode, value: Value) => listener(anchorNode, value),
+				);
+				const unsubscribeFromChildrenChange = this.anchorNode.on(
+					"childrenChanging",
+					(anchorNode: AnchorNode) => listener(anchorNode, undefined),
+				);
+				return () => {
+					unsubscribeFromValueChange();
+					unsubscribeFromChildrenChange();
+				};
+			}
+			case "subtreeChanging": {
+				const unsubscribeFromSubtreeChange = this.anchorNode.on(
+					"subtreeChanging",
+					(anchorNode: AnchorNode) => {
+						return (listener as EditableTreeEvents["subtreeChanging"])(anchorNode);
+					},
+				);
+				return () => {
+					unsubscribeFromSubtreeChange();
+				};
```

Comment:
I think this would be slightly more efficient and simpler code.
Though if you want to keep it as is for consistency with the other code thats fine as well.
```suggestion
				return unsubscribeFromSubtreeChange;
```

Reply:
Looks good to me. Will perform the change as suggested.